### PR TITLE
Fix deleting multiple items

### PR DESCRIPTION
**Pull request by Sjoerd1993**
In the latest build, only the top item gets deleted when pressing the delete button. The reason for this is that it loops through the dictionary itself in actions.py, changing the dictionary size upon iteration which is not allowed. This solves this by looping through the selected keys instead like in previous releases.

### CODE REVIEW

**Review by cmkohnen — CHANGES_REQUESTED** [review]
Ah, the reason, I reformatted this is to have on less iteration.
Using your approach we iterate twice.
However I found that just inserting a `.copy()` like the following works just as well and resorts to only one iteration:
`for item in self.datadict.copy().values()`

**Comment by Sjoerd1993** [comment]
> Using your approach we iterate twice.

Yeah, once over the entire dictionary to retrieve the selected keys, and then once more over each selected key to delete them. Making a copy of the datadict works indeed without any issues, and keeps everything contained to a single iteration.

It's an interesting hack actually, because you are deleting the items in your copy. But because Python doesn't create an actual copy, but refers to the same object in memory when copying dictionaries, it deletes the original item as well. I honestly didn't expect that would work (as both the copy and the original should change size in iteration right?), but somehow it does.  

It's a fine solution for me. As it's not a deep copy, it won't really change memory consumption by making this copy either.

**Comment by cmkohnen** [comment]
It behaves actually a bit different. The copied dict will stay the same size and should get picked up by the garbage collector when the function ends. Since items contain their own key, we can reference the actual item to be deleted, as the graphs.delete() function takes only a key

File: src/actions.py [hunk 1]
```diff
@@ -117,11 +117,10 @@ def on_response(_dialog, response):
 
 
 def delete_selected_action(_action, _target, self):
-    for item in self.datadict.values():
+    for item in self.datadict.copy().values():
         if item.selected:
             graphs.delete_item(self, item.key, True)
 
-
```
Comment (cmkohnen) [inline src/actions.py:124]: don't know why flake didn't pick up on this: two newlines required here

**Comment by Sjoerd1993** [comment]
> the graphs.delete() function takes only a key

Somehow in my head I was thinking we were deleting the copy on each loop, taking away the original with it. The deletion is indeed done on the actual item using the key. Either way, it does create a new dictionary in memory (at least within the life-span of the function), meaning the original will remain its size anyway. 

I got things mixed up with regular assignments where both dictionaries refer to the same points in memory I've had some issues with regular shallow copies in the past, where I was probably thinking of. Either way, the point is that you are correct.

Thanks for the elaboration.

**Review by cmkohnen — APPROVED** [review]

**Pull request merged**